metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -302,20 +303,39 @@ func (sink *StackdriverSink) requestSender(reqQueue chan *sd_api.CreateTimeSerie
 	}
 }
 
+func marshalRequestAndLog(printer func([]byte), req *sd_api.CreateTimeSeriesRequest) {
+	reqJson, errJson := json.Marshal(req)
+	if errJson != nil {
+		printer(reqJson)
+	} else {
+		glog.Errorf("Couldn't marshal Stackdriver request %v", errJson)
+	}
+}
+
 func (sink *StackdriverSink) sendOneRequest(req *sd_api.CreateTimeSeriesRequest) {
 	startTime := time.Now()
 	empty, err := sink.stackdriverClient.Projects.TimeSeries.Create(fullProjectName(sink.project), req).Do()
 
 	var responseCode int
 	if err != nil {
 		glog.Warningf("Error while sending request to Stackdriver %v", err)
+		if glog.V(2) {
+			marshalRequestAndLog(func(reqJson []byte) {
+				glog.Warningf("The request was: %s", reqJson)
```

Comment:
glog.Infof.V(2)

Reply:
There is no such syntax. There is glog.V(2).Infof, however. Quoting from glog doc, those two calls are (almost) equivalent:

	if glog.V(2) { glog.Info("log this") }

	glog.V(2).Info("log this")

The only difference is, when V(2) is false, the expression inside won't get evaluated. I'm using the former, since instead of simply logging a string I'm also lazy-evaluating conversion to json.

Reply:
Sorry, I was not clear enough. When you are reading the code IMO it's easier to understand:
```
if glog.V(2) {
  glog.V(2).Info("log this") }
}
```
than
```
if glog.V(2) {
  glog.Warninf("log this") }
}
```
because in the second case your first impression is that the log severity is `Warning`, but then when you look closer, it turns out the this is in fact `V(2)`.

Reply:
Ack, changed.